modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -870,6 +870,25 @@ def get_indices(cls, axis, partitions, index_func=None):
         # TODO FIX INFORMATION LEAK!!!!1!!1!!
         return new_idx[0].append(new_idx[1:]) if len(new_idx) else new_idx
 
+    @classmethod
+    def _update_partition_dimension_caches(cls, partitions: np.ndarray):
+        """
+        Build and set the length and width caches of each _physical_ partition in the given list of partitions.
+
+        Parameters
+        ----------
+        partitions : np.ndarray
+            The partitions for which to update length caches.
+
+        Notes
+        -----
+        For backends that support parallel computations, these caches should be computed asynchronously.
```

Comment:
nit: "these caches *are* computed...". IMO this is clearer because sometimes when we say "should", we mean "should theoretically" or "should eventually (when I get around to implementing it)".

Reply:
changed, in my head "should" implies that it is a contract for implementors to follow, but I'll adhere to the standard of the codebase